apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -43,28 +42,14 @@ const PageLinkElement = ({ value, onChange }: PageLinkElementProps) => {
 
   const selectedResourceId = getResourceIdFromReferenceLink(value)
 
-  const { data: resource } = trpc.resource.getWithFullPermalink.useQuery({
-    resourceId: selectedResourceId,
-  })
-
   return (
-    <>
-      <ResourceSelector
-        siteId={String(siteId)}
-        onChange={(resourceId) =>
-          onChange(getReferenceLink({ siteId: String(siteId), resourceId }))
-        }
-        selectedResourceId={selectedResourceId}
-      />
-
-      {!!resource && (
-        <Box bg="utility.feedback.info-subtle" p="0.75rem" w="full" mt="0.5rem">
-          <Text textStyle="caption-1">
-            You selected /{resource.fullPermalink}
-          </Text>
-        </Box>
-      )}
-    </>
+    <ResourceSelector
+      queryFn={trpc.resource.getChildrenOf.useInfiniteQuery}
```

Comment:
nit: to be honest, this feels like a code-smell. this works for now because they have relatively similar interfaces but we tie it very closely to the actual function too (the type of `queryFn` is specified as those 2 queries). i worry that if change is needed to either of the 2 functions, this component might not be as easily reusable down the line.

something we could consider is shifting the subcomponents out and favouring composition so that, for example, 1 component purely displays another and another triggers the refetch. another alternative we could do is specify the type ourselves + specify the refetch etc as extra props so we don't depend so strictly on teh trpc type!

Reply:
i thought about it for some time and decided to update it to pass in a prop `onlyShowFolders` which will be used to decide which query to used

```ts
  const queryFn = onlyShowFolders
    ? trpc.resource.getFolderChildrenOf.useInfiniteQuery
    : trpc.resource.getChildrenOf.useInfiniteQuery
```

Why this is an improvement:
- **Centralized Updates**: If there are changes to the `getFolderChildrenOf` or `getChildrenOf` props, we only need to update them in a single location (this file). Most of the changes will be confined to updating the props, which simplifies maintenance and enhances extensibility without over-engineering the solution.
- **Flexibility in Handling Query Changes**: If the interfaces of the queries change significantly, we can simply create a copy and preserve the original query. This approach avoids forcing them to reuse the same query, providing more flexibility as the system evolves.
- **Simplified Parent Component Props**: The parent component no longer needs to be aware of which specific query to pass in. This reduces complexity and decouples the parent from needing to handle the specifics of the query logic.